libr/parse/p/parse_arm_pseudo.c

```diff
@@ -324,26 +324,39 @@ static bool varsub(RParse *p, RAnalFunction *f, ut64 addr, int oplen, char *data
 	bool ucase = IS_UPPER (*tstr);
 	RAnalVarField *var;
 	r_list_foreach (bpargs, iter, var) {
-		char *reg = anal->reg->name[R_REG_NAME_BP];
 		char *tmplt = NULL;
-		if (var->delta > -10 && var->delta < 10) {
+		st64 delta = ST64_MAX;
+		if (p->get_ptr_at) {
+			delta = p->get_ptr_at (f, var->delta, addr);
+		}
+		if (delta == ST64_MAX && var->field) {
+			delta = var->delta + f->bp_off;
+		} else if (delta == ST64_MAX) {
+			continue;
+		}
+		const char *reg = NULL;
+		if (p->get_reg_at) {
+			reg = p->get_reg_at (f, var->delta, addr);
+		}
+		if (!reg) {
+			reg = anal->reg->name[R_REG_NAME_BP];
+		}
+		if (delta > -10 && delta < 10) {
 			if (p->pseudo) {
 				char sign = '+';
-				int delta = var->delta;
-				if (var->delta < 0) {
+				if (delta < 0) {
 					sign = '-';
-					delta = -delta;
 				}
-				oldstr = r_str_newf ("[%s %c %d]", reg, sign, delta);
+				oldstr = r_str_newf ("[%s %c %d]", reg, sign, R_ABS (delta));
 			} else {
-				oldstr = r_str_newf ("[%s, %d]", reg, var->delta);
+				oldstr = r_str_newf ("[%s, %d]", reg, delta);
```

Comment:
uhm, not R_ABS here?

Reply:
@trufae no, thats signed, the R_ABS above is only to get the `+` and a separated `-` on pseudo